.github/workflows/macos.yml

```diff
@@ -28,7 +28,9 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Update submodules
-        run: git submodule update --init --recursive
+        run: |
+          git submodule sync --recursive
```

Comment:
I think `actions/checkout` automtaically checkout the submodule recursively: 

```
  /usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
```

Reply:
I guess a better approach would be 

```
      - name: Check out source repository
        uses: actions/checkout@v2
        with:
            submodules: recursive
```

similar to https://github.com/facebookresearch/torcharrow/blob/6737648fdd081f6b5a46ce0d041dca3e80ea50f1/.github/workflows/nightly-wheel.yml#L27-L30

Reply:
yes we used this for nightly build..